### PR TITLE
add the possibility to add display mode dependant layout handles

### PR DESCRIPTION
**Pull request by brosenberger**
### Description (*)
layout updates for categories can only be automatically added by default handle, pagetype (anchor, with children,...), with id and custom settings stored within the category.
Another option would be to set the display mode within the backend (static block, static block + products or products only) which is not considered for layout handles from the controller. this commit adds the possibility to add displaymode dependant layout handles:

* catalog_category_view_displaymode_page.xml (Static block only)
* catalog_category_view_displaymode_products.xml (Products only)
* catalog_category_view_displaymode_products_and_page.xml (Static block and products)

### Manual testing scenarios (*)

1. add a display mode specific layout xml
2. set a catgory to that specific display mode
3. clear caches
4. go to the changed category and test if the newly added blocks/container/... from the specific layout xml are applied

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Review by aleron75 — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @aleron75, thank you for the review.
[ENGCOM-6325](https://jira.corp.magento.com/browse/ENGCOM-6325) has been created to process this Pull Request
:eight_spoked_asterisk: @aleron75, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Alfa** [comment]
Hi @brosenberger  .
     Due to [Magento Definition of Done](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing_dod.html) all code must be covered by tests. Please cover your fix by unit test.
Thanks!

**Comment by engcom-Golf** [comment]
I will take care of test coverage

**Comment by engcom-Golf** [comment]
HI @aleron75 all changes covered by unit test, could you please review ?

File: app/code/Magento/Catalog/Test/Unit/Controller/Category/ViewTest.php [hunk 1]
```diff
@@ -199,11 +205,45 @@ public function testApplyCustomLayoutUpdate()
...
+    }
+
+    /**
+     * Data provider for execute method.
+     *
+     * @return array
+     */
+    public function getInvocationData(): array
+    {
+        return [
```
Comment (aleron75) [inline app/code/Magento/Catalog/Test/Unit/Controller/Category/ViewTest.php:250]: hi, would you consider expanding the test to check different page types and display modes?

Comment (engcom-Golf) [reply app/code/Magento/Catalog/Test/Unit/Controller/Category/ViewTest.php:250]: Yes sure :+1:

**Comment by engcom-Golf** [comment]
@aleron75 all done :heavy_check_mark:

**Review by aleron75 — APPROVED** [review]
well done!

**Comment by magento-engcom-team** [comment]
Hi @aleron75, thank you for the review.
[ENGCOM-6325](https://jira.corp.magento.com/browse/ENGCOM-6325) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
@magento run all tests

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Pull request merged**